### PR TITLE
[SPARK-25186][SQL] Remove v2 save mode.

### PR DESCRIPTION
**Pull request by rdblue**
## What changes were proposed in this pull request?

This removes `SaveMode` from the v2 write API. Overwrite is temporarily implemented by deleting path-based and name-based tables and appending. These appends implicitly create tables and should be replaced with CTAS and RTAS.

## How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
@rdblue, is it a mistake to include the commits for SPARK-25188?

**Comment by rdblue** [comment]
@HyukjinKwon, those changes probably don't need to be in this PR, but this is just a demonstration that we can remove `SaveMode` without changing test cases. The larger issue is that this doesn't correctly use CTAS or RTAS plans. Instead, it does things like directly deleting data.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala [hunk 1]
```diff
@@ -143,14 +145,17 @@ object DataSourceV2Strategy extends Strategy {
...
-      WriteToDataSourceV2Exec(writer, planLater(query)) :: Nil
+    case WriteToDataSourceV2(writer, writeConfig, query) =>
+      WriteToDataSourceV2Exec(writer, writeConfig, planLater(query)) :: Nil
 
     case AppendData(r: DataSourceV2Relation, query, _) =>
-      WriteToDataSourceV2Exec(r.newWriteSupport(), planLater(query)) :: Nil
+      val writeSupport = r.newWriteSupport()
+      val writeConfig = writeSupport.createWriteConfig(
+        query.schema, new DataSourceOptions(r.options.asJava))
+      WriteToDataSourceV2Exec(r.newWriteSupport(), writeConfig, planLater(query)) :: Nil
```
Comment (viirya) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala]: Why do `r.newWriteSupport()` again? We can reuse `writeSupport`.

Comment (rdblue) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala]: Looks like an accident to me.

**Pull request closed without merging**